auth/realm/base/src/test/java/org/wildfly/security/auth/realm/AggregateRealmEvidenceTest.java

```diff
@@ -0,0 +1,336 @@
+ * limitations under the License.
+ */
+
+package org.wildfly.security.auth.realm;
+
+import java.io.File;
+import java.io.IOException;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.FileVisitResult;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.nio.file.SimpleFileVisitor;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.Principal;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.Locale;
+import java.util.function.Function;
+import javax.security.auth.x500.X500Principal;
+import org.junit.Assert;
+import org.junit.Test;
+import org.wildfly.security.asn1.ASN1Encodable;
+import org.wildfly.security.auth.principal.NamePrincipal;
+import org.wildfly.security.auth.server.SecurityRealm;
+import org.wildfly.security.auth.server.EvidenceDecoder;
+import org.wildfly.security.auth.server.NameRewriter;
+import org.wildfly.security.auth.server.RealmIdentity;
+import org.wildfly.security.auth.server.ModifiableRealmIdentity;
+import org.wildfly.security.authz.Attributes;
+import org.wildfly.security.authz.MapAttributes;
+import org.wildfly.security.evidence.Evidence;
+import org.wildfly.security.evidence.X509PeerCertificateChainEvidence;
+import org.wildfly.security.x500.GeneralName;
+import org.wildfly.security.x500.X500;
+import org.wildfly.security.x500.X500AttributeTypeAndValue;
+import org.wildfly.security.x500.X500PrincipalBuilder;
+import org.wildfly.security.x500.cert.SubjectAlternativeNamesExtension;
+import org.wildfly.security.x500.cert.X509CertificateBuilder;
+
+/**
+ * Test case testing the {@link AggregateSecurityRealm} implementation.
+ *
+ * @author <a href="mailto:aabdelsa@redhat.com">Ashley Abdel-Sayed</a>
+ */
+public class AggregateRealmEvidenceTest {
+    private static final String IDENTITY_NAME = "CN=bob0";
+    private static final String IDENTITY_NAME_TRANSFORMED = "CN=BOB0";
+    private static final Principal IDENTITY_PRINCIPAL = new NamePrincipal(IDENTITY_NAME);
+    private static final Principal IDENTITY_PRINCIPAL_TRANSFORMED = new NamePrincipal(IDENTITY_NAME_TRANSFORMED);
+
+
+    /*
+     * The intent of this test case is to focus on the realm aggregation aspect, the AggregateAttributesTest focuses
+     * on different permutations of actual attribute aggregation. This test focuses on using evidence to obtain the identity,
+     * the AggregateRealmTest focuses on using a principal to obtain the identity.
+     */
+
+    @Test
+    public void testAuthenticationOnly() throws Exception {
+        Attributes authenticationAttributes = new MapAttributes();
+        authenticationAttributes.add("team", 0, "One");
+        X509PeerCertificateChainEvidence evidence = new X509PeerCertificateChainEvidence(populateCertificateChain(true));
+        evidence.setDecodedPrincipal(IDENTITY_PRINCIPAL);
```

Comment:
I think the ```setDecodedPrincipal``` calls can be removed since the default principal for this type of evidence is the subject name (which is what ```IDENTITY_PRINCIPAL``` is).